ui/selectors/approvals.ts

```diff
@@ -1,19 +1,71 @@
 import { ApprovalControllerState } from '@metamask/approval-controller';
 import { ApprovalType } from '@metamask/controller-utils';
+import { transactionMatchesNetwork } from '../../shared/modules/transaction.utils';
```

Comment:
If we're trying to isolate the approval selectors in a new file, I'd recommend not importing any of these transaction helpers to avoid coupling it to the transaction logic.

We could either move the entire new selector out, or add a callback argument such as `matchesCurrentChain`.

Reply:
Moved out the selector itself.